Lib/asyncio/tasks.py

```diff
@@ -53,7 +53,10 @@ def current_task(cls, loop=None):
         """
         if loop is None:
             loop = events.get_event_loop()
-        return cls._current_tasks.get(loop)
+        try:
+            return loop.current_task()
+        except (AttributeError, NotImplementedError):
+            return cls._current_tasks.get(loop)
```

Comment:
Let's rewrite as:

```python
try:
    current_task_meth = loop.current_task
except AttributeError:
    pass
else:
    try:
        return current_task_meth()
    except NotImplementedError:
        pass
```

This way we won't accidentally trap an `AttributeError` from a badly behaving `loop.current_task()`.

Reply:
BTW, don't forget to update `_asynciomodule.c` to use new loop methods too.